lib/pushmi_pullyu/swift_depositer.rb

```diff
@@ -1,37 +1,46 @@
 require 'digest/md5'
 require 'openstack'
-
 class PushmiPullyu::SwiftDepositer
 
-  attr_reader :swift_connection
-
-  def initialize(connection)
+  def initialize(connection, connection_wait_time: 10)
+    @connection_wait_time = connection_wait_time
     # Generic authentication parameters
-    swift_connection_parameters = {
+    @swift_connection_parameters = {
       username: connection[:username],
       api_key: connection[:password],
       auth_url: connection[:auth_url],
       project_name: connection[:project_name],
       auth_method: 'password',
-      service_type: 'object-store'
+      service_type: 'object-store',
+      # The retry_auth value should be true by default. It is currently set
+      # to nil which causes the connection to die without giving more error
+      # details.
+      retry_auth: true
     }
 
     if connection[:auth_version] == 'v3'
-      swift_connection_parameters[:user_domain] = connection[:user_domain]
+      @swift_connection_parameters[:user_domain] = connection[:user_domain]
     elsif connection[:auth_version] == 'v1'
-      swift_connection_parameters[:project_domain_name] = connection[:project_domain_name]
-      swift_connection_parameters[:authtenant_name] = connection[:tenant]
+      @swift_connection_parameters[:project_domain_name] = connection[:project_domain_name]
+      @swift_connection_parameters[:authtenant_name] = connection[:tenant]
     end
 
-    @swift_connection = OpenStack::Connection.create(swift_connection_parameters)
+    establish_connection
   end
 
   def deposit_file(file_name, swift_container)
     file_base_name = File.basename(file_name, '.*')
-
     checksum = Digest::MD5.file(file_name).hexdigest
-
-    era_container = swift_connection.container(swift_container)
+    begin
+      era_container = @swift_connection.container(swift_container)
+    rescue OpenStack::Exception::ExpiredAuthToken => e
```

Comment:
With `@retry_auth` true is this redundant?

Seems similar to https://github.com/ruby-openstack/ruby-openstack/blob/373e0c47ab23b67c695873839282866d0849989a/lib/openstack/connection.rb#L203-L207

Reply:
You are right. However, I could test that exception in our code, not the OpenStack gem, so I left our retry.

I am worried about the gem not being maintained. This us causing an error at the moment.

I am open to removing my code that catches the exception in order to have less things to maintain on our side. Let me know what you think.